muspinsim_plot/muspinsim_plot.xml

```diff
@@ -0,0 +1,105 @@
+       python ${__tool_directory__}/generate_plot.py inputs.json
+    ]]></command>
+    <configfiles>
+        <inputs name="inputs" data_style="paths" filename="inputs.json" />
+    </configfiles>
+    <inputs>
+        <param type="text" name="title" label="Plot Title" value="" />
+        <param type="text" name="xlab" label="Label for x axis" value="" />
+        <param type="text" name="ylab" label="Label for y axis" value="" />
+        <param name="outfile_type" type="select" label="Output File Type" >
+            <option value="pdf">PDF</option>
+            <option value="png">PNG</option>
+        </param>
+        <repeat name="mu_out_series" title="Muspinsim File Series">
+            <param type="data" name="mu_data" label="Muspinsim Experiment Data (.dat)" />
+            <param type="text" name="mu_label" optional="true" label="Muspin Experiment Label" value="" />
+            <param type="select" name="colour" label="Line/Point Color">
+                <option value="k">Black</option>
+                <option value="r">Red</option>
+                <option value="g">Green</option>
+                <option value="b">Blue</option>
+            </param>
+            <conditional name="series_type">
+                <param type="select" name="type" label="Series Type" >
+                    <option value="line">Line</option>
+                    <option value="points">Points</option>
+                </param>
+                <when value="line">
+                   <param type="select" name="linestyle" label="Line type">
+                       <option value="solid">Solid</option>
+                       <option value="dashed">Dashed</option>
+                       <option value="dotted">Dotted</option>
+                   </param>
+                   <param type="float" name="linewidth" label="Line width" value="1.0"/>
+                </when>
+                <when value="points">
+                    <param type="select" name="pointstyle" label="Point Type">
+                        <option value="s">Square</option>
+                        <option value="o">Circle</option>
+                    </param>
+                    <param type="float" name="pointscale" label="Point Scale" value="1.0"/>
+                </when>
+            </conditional>
+        </repeat>
+    </inputs>
+    <outputs>
+        <data format="pdf" name="out_file_pdf" from_work_dir="outfile.pdf" >
+            <filter>(outfile_type == 'pdf')</filter>
+        </data>
+        <data format="png" name="out_file_png" from_work_dir="outfile.png">
+            <filter>(outfile_type == 'png')</filter>
+        </data>
+    </outputs>
+    <tests>
+        <test>
+            <param name="title" value="Temperature Example"/>
+            <param name="xlab" value="Time"/>
+            <param name="ylab" value="Asymmetry"/>
+            <param name="outfile_type" value="png"/>
+            <param name="mu_data" value="high_T.dat" ftype="txt"/>
```

Comment:
Is this name equivalent to the name in line 84, or does Galaxy handle this a different way?

Reply:
They are equivalent, but I found that it was easier to just duplicate params in tests and galaxy automatically assigns them to appropriate indexes, see [d74459d](https://github.com/muon-spectroscopy-computational-project/muon-galaxy-tools/pull/55/commits/d74459dccb32daa6a304f0cd5c503d4db1cfd900)

Reply:
New commit also fixes an issue where test wasn't picking up the correct parameters and instead using defaults. (like not picking up the correct colour or linewidth etc)